packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -48,11 +55,27 @@ export const withReactQuery = (Wrapped) => {
             this.props.locals.__queryClient =
                 this.props.locals.__queryClient || new QueryClient(queryClientConfig)
 
+            const optionsForSSRAndHydration = {
+                ...queryClientConfig.defaultOptions,
+                queries: {
+                    ...queryClientConfig.defaultOptions?.queries,
+                    // Stop queries during hydration because they've run already on server side
+                    ...(!isServerSide ? {refetchOnMount: false} : {}),
+                    // On server side, make sure query with error does not attempt to fetch again on mount
+                    // NOTE: `retryOnMount` and `retry` options are different. They seem to work independently.
+                    ...(isServerSide ? {retryOnMount: false} : {})
```

Comment:
I didn't talk about this in the PR description. But it's related to and fixes the issue for this [other ticket](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017x6qOYAQ/view).

How to verify it? 
1. `npm start` in the test-commerce project
2. Navigate to http://localhost:3000/query-errors?mobify_server_only
3. Verify that the `isLoading` state is now correctly `false`.

What happened is previously the `isLoading` state would still be true. The reason is because by default, `retryOnMount` is true. So whenever there's an error with the query, react-query would want to fetch again on component mount. The SSR period ends as the query is fetching again → that's why `isLoading` has turned to true.